src/applications/vaos/tests/new-appointment/components/PreferredDatePage.unit.spec.jsx

```diff
@@ -1,133 +1,145 @@
+      store,
+    });
+
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /month/i }), [
+      '2',
+    ]);
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /day/i }), [
+      '2',
+    ]);
+    await userEvent.type(
+      screen.getByRole('spinbutton', { name: /year/i }),
+      '2050',
     );
-
-    form.find('form').simulate('submit');
-
-    expect(form.find('.usa-input-error').length).to.equal(1);
-    expect(history.push.called).to.be.false;
-    form.unmount();
+    fireEvent.click(screen.getByText(/Continue/));
+    expect(await screen.findByRole('alert')).to.contain.text(
+      'Please enter a date less than 395 days in the future ',
+    );
+    expect(screen.history.push.called).to.be.false;
   });
 
-  it('should submit with valid data', () => {
-    const maxDate = moment()
+  it('should submit with valid data', async () => {
+    const maxMonth = moment()
       .add(395, 'days')
-      .format('YYYY-MM-DD');
-
-    const openFormPage = sinon.spy();
-    const routeToNextAppointmentPage = sinon.spy();
-
-    const form = mount(
-      <PreferredDatePage
-        openFormPage={openFormPage}
-        routeToNextAppointmentPage={routeToNextAppointmentPage}
-        data={{ preferredDate: maxDate }}
-      />,
+      .format('MM');
+    const maxDay = moment()
+      .add(395, 'days')
+      .format('DD')
+      .replace(/\b0/g, '');
+    const maxYear = moment()
+      .add(395, 'days')
+      .format('YYYY');
+
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(<PreferredDatePage />, {
+      store,
+    });
+
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /month/i }), [
+      maxMonth,
+    ]);
+    userEvent.selectOptions(screen.getByRole('combobox', { name: /day/i }), [
+      maxDay,
+    ]);
+    await userEvent.type(
+      screen.getByRole('spinbutton', { name: /year/i }),
+      maxYear,
     );
-
-    form.find('form').simulate('submit');
-
-    expect(form.find('.usa-input-error').length).to.equal(0);
-    expect(routeToNextAppointmentPage.called).to.be.true;
-    form.unmount();
+    fireEvent.click(screen.getByText(/Continue/));
+    expect(screen.history.push.called).to.be.true;
   });
 
   it('document title should match h1 text', () => {
-    const openFormPage = sinon.spy();
-    const updateFormData = sinon.spy();
-    const pageTitle = 'Tell us when you want to schedule your appointment';
-
-    const form = mount(
-      <PreferredDatePage
-        openFormPage={openFormPage}
-        updateFormData={updateFormData}
-        data={{}}
-      />,
-    );
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(<PreferredDatePage />, {
+      store,
+    });
 
-    expect(form.find('h1').text()).to.equal(pageTitle);
-    expect(document.title).contain(pageTitle);
-    form.unmount();
+    expect(screen.baseElement).to.contain.text(
```

Comment:
I think I'd put this expect in one of the other tests and remove this separate test.

Reply:
I agree. Changing now.